snapcraft/cli/_runner.py

```diff
@@ -52,6 +53,38 @@
 ]
 
 
+def _is_legacy_reexec() -> bool:
```

Comment:
I think `_is_legacy` would read better, the function itself doesn't have anything to do with re-execing.